tests/common/plugins/ptfadapter/ptfadapter.py

```diff
@@ -88,3 +91,27 @@ def reinit(self, ptf_config=None):
         """
         self.kill()
         self._init_ptf_dataplane(self.ptf_ip, self.ptf_nn_port, self.device_num, self.ptf_port_set, ptf_config)
+
+    def update_payload(self, pkt):
+        if isinstance(pkt, scapy.Ether):
+            for proto in (scapy.UDP, scapy.TCP):
+                if proto in pkt:
+                    pkt[proto].load = self._update_payload(pkt[proto].load)
+        elif isinstance(pkt, mask.Mask):
+            for proto in (scapy.UDP, scapy.TCP):
+                if proto in pkt.exp_pkt:
+                    pkt.exp_pkt[proto].load = self._update_payload(pkt.exp_pkt[proto].load)
+        return pkt
+
+    def _update_payload(self, payload):
+        if self.payload_pattern:
+            len_old = len(payload)
+            len_new = len(self.payload_pattern)
+            if len_new >= len_old:
+                return self.payload_pattern[:len_old]
+            else:
+                factor = len_old/len_new + 1
+                new_payload = self.payload_pattern * factor
+                return new_payload[:len_old]
+        else:
+            return payload
```

Comment:
How does this work for test cases where we care about the payload (e.g. maybe we're carrying an HTTP request or verifying some other protocol that is built on top of TCP/UDP or something)?

Reply:
It won't work if we care about the payload. Then it is not a good idea to tamper the packets to be injected.

Reply:
I'll add an option for skipping packet tampering. In case some test need the payload to be untouched, then that test case can set the option to skip it.

Reply:
Sounds good! I think this is a good default to have, but we definitely need to be able to bypass for some test scenarios.

Reply:
Done. Added an option to bypass the packet payload updating behavior.